### PR TITLE
Study helper phase 2 update 

### PR DESCRIPTION
**Pull request by iDasher**
I implemented 1-3 from the phase 2 issue, meaning I now can test the user on their questions, give a score and show them their performance compared to prior attempts at that quiz to help see if they are improving. Additionally added javadoc to many methods and more testing.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -56,3 +56,6 @@ bin/
 
 ### Mac OS ###
 .DS_Store
+
+### Storing files ###
```
Comment (xinyue296) [inline .gitignore]: is this the correct path?

**Review by xinyue296 — CHANGES_REQUESTED** [review]
Thank you for the hard work! Good job on testing!

**Review by Overpoweredplatypus — APPROVED** [review]
Excellent

**Review by xinyue296 — APPROVED** [review]

**Review by sailoraria — APPROVED** [review]
Looks good!

**Pull request merged**